### PR TITLE
Trailing limit or ; schema inference bug quick fix

### PR DESCRIPTION
**Pull request by mpeaton**

### CODE REVIEW

File: ibis/mapd/client.py [hunk 1]
```diff
@@ -860,7 +860,15 @@ def sql(self, query):
         """
         # Get the schema by adding a LIMIT 0 on to the end of the query. If
         # there is already a limit in the query, we find and remove it
-        limited_query = 'SELECT * FROM ({}) t0 LIMIT 1'.format(query)
```
Comment (cpcloud) [inline ibis/mapd/client.py]: Is the issue with this that omnisci cannot handle nested `LIMIT` queries?

Comment (cpcloud) [reply ibis/mapd/client.py]: Or is the compilation incorrect? If the latter then we should fix the code gen rather than use regex substitution.

Comment (mpeaton) [reply ibis/mapd/client.py]: As far as I can discern, yes.  Nested LIMIT queries are failing with the following error. 
``` Exception: Sort node not supported as input to another sort```
 We are considering exposing a Validator, and improved schema inference for this backend, but it's a WIP.

Comment (cpcloud) [reply ibis/mapd/client.py]: Can you add a test that fails without this change and passes with it? It'd be nice to prevent a regression.

Comment (xmnlab) [reply ibis/mapd/client.py]: maybe the test should check also cases like (e.g.):

```
queries = [
        'select * from functional_alltypes limit 1, 2',
        'select * from functional_alltypes limit \n  1, 2',
        'select * from functional_alltypes limit \t  1, 2\n  ',
    ]
```

Comment (mpeaton) [reply ibis/mapd/client.py]: Yes absolutely.   Is there currently a test class or set  to test edge cases of SQL validity?

**Comment by xmnlab** [comment]
@mpeaton it seems #1903 continues the work you start here,  could we close this PR?

**Comment by mpeaton** [comment]
This PR is handled in part by #1903 . it is expected that another PR will be started to address SQL string validity as needed.
